sriov: Do not resolve sriov refer in `gen_diff`

cathay4t opened the pull request:
When using `nmstatectl apply` for this YAML:

```yml
---
interfaces:
- name: sriov:ens7f1np1:3
  type: ethernet
  ipv4:
    enabled: false
- name: ens7f1np1
  type: ethernet
  ethernet:
    sr-iov:
      total-vfs: 6
```

Nmstate correctly applied the desired state, but got error in
`gen_diff()`:

   InvalidArgument: Failed to find SR-IOV VF interface name for
  sriov:ens7f1np1:3

This is because SRIOV VF information does not exists in pre-apply current
state used for generating this diff state.

The fix is do not resolve SRIV VF reference in `gen_diff` mode.

Unit test case included.

Resolves: https://issues.redhat.com/browse/RHEL-91588

cathay4t committed:
cli: Do not fail the apply action on `gen_diff` error

Even when apply action succeeded, the failure of `gen_diff` will cause
`nmstatectl apply` fails but the desired state is actually applied.

Signed-off-by: Gris Ge <fge@redhat.com>

ihuguet commented on rust/src/cli/apply.rs:
```diff
@@ -265,7 +265,15 @@ async fn apply_state_async(
+            match net_state.gen_diff(&cur_state) {
+                Ok(s) => Ok(s),
+                Err(e) => {
+                    log::warn!(
+                        "Failed to generate difference: {e}, \
+                        returning full desired state");
+                    Ok(net_state.clone())
```
I'd print at error level prefixed with "Bug:" because it is a bug in the code.

cathay4t replied:
Changed to error with `BUG: ` prefix.

cathay4t committed:
sriov: Do not resolve sriov refer in `gen_diff`

When using `nmstatectl apply` for this YAML:

```yml
interfaces:
- name: sriov:ens7f1np1:3
  type: ethernet
  ipv4:
    enabled: false
- name: ens7f1np1
  type: ethernet
  ethernet:
    sr-iov:
      total-vfs: 6
```

Nmstate correctly applied the desired state, but got error in
`gen_diff()`:

    InvalidArgument: Failed to find SR-IOV VF interface name for
    sriov:ens7f1np1:3

This is because SRIOV VF information does not exists in pre-apply
current state used for generating this diff state.

The fix is do not resolve SRIV VF reference in `gen_diff` mode.

Unit test case included.

Resolves: https://issues.redhat.com/browse/RHEL-91588

Signed-off-by: Gris Ge <fge@redhat.com>
